packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  isXMLNamespaceKey,
+  flattenProperties,
+  flattenEvents,
+  flattenAssociations,
+  flattenAggregations,
+  ui5NodeToFQN,
+} from "@ui5-language-assistant/logic-utils";
+import { UnknownAttributeKeyIssue } from "../../../api";
+import { TEMPLATING_NS, CUSTOM_DATA_NS } from "../../utils/special-namespaces";
+
+export function validateUnknownAttributeKey(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UnknownAttributeKeyIssue[] {
+  if (attribute.syntax.key === undefined || attribute.key === null) {
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  if (isAttributeNameAlwaysValid(attribute)) {
+    return [];
+  }
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  let ui5Aggregation: UI5Aggregation | undefined = undefined;
+  if (ui5Class === undefined) {
+    ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  }
+
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  // Special case: "class" property can be used on all Controls and selected Elements.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "class") {
+    return [];
+  }
+
+  // Special case: "binding" property is a synonym for "objectBindings" special setting on ManagedObject
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "binding") {
+    return [];
+  }
+
+  // Special case: "stashed" property can appear under sap.uxap.ObjectPageLazyLoader
+  // Technically it can be under any Element but it should only be used for this one.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (
+    attribute.key === "stashed" &&
+    ui5NodeToFQN(ui5Class) === "sap.uxap.ObjectPageLazyLoader"
+  ) {
+    return [];
+  }
+
+  const allProps: (
+    | UI5Prop
+    | UI5Event
+    | UI5Association
+    | UI5Aggregation
+  )[] = flattenProperties(ui5Class);
+  const allEvents = flattenEvents(ui5Class);
+  const allAssociations = flattenAssociations(ui5Class);
+  // Aggregations can be used as attributes for binding
+  const allAggregations = flattenAggregations(ui5Class);
+  const allClassUI5Attributes = allProps
+    .concat(allEvents)
+    .concat(allAssociations)
+    .concat(allAggregations);
+  // TODO Should we filter out non-public/protected names?
```

Comment:
I naively think usage of private/hidden/top-secret visibility is a **different** validation.

Reply:
Since the error should be the same according to Frank (they give the same error in runtime) I think it's ok to keep it here. If we need to differentiate them in the future we can extract it to a different validation.

Reply:
normally I would think these are different errors, but if Frank says it should be the same error its fine.

Afterall UI5 private !== TypeScript private.